tests/test_parser.py

```diff
@@ -142,8 +142,7 @@ class TestParser (unittest.TestCase):
     def test_parse (self, _in, _out):
         # Parse all test patterns in one go.
         out = StringIO()
-        handler = HtmlPrettyPrinter(out)
-        htmlsoup.process_soup(handler, htmlsoup.make_soup(_in))
+        HtmlPrettyPrinter(out, htmlsoup.make_soup(_in))
```

Comment:
Here too: we have a class that seemingly does nothing.  It would be fine if it were a function `pretty_print_htm(out, htmlsoup.make_soup(_in))`, but make the name upper-case and it looks wrong.

Reply:
Yes, it only needs to be a function now and not a class - happy to take naming suggestions like that!